android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewShadowNode.kt

```diff
@@ -65,20 +66,20 @@ class SafeAreaViewShadowNode : LayoutShadowNode() {
     left = PixelUtil.toPixelFromDIP(left)
     val edges = localData.edges
     val insets = localData.insets
-    val insetTop: Float = if (edges.contains(SafeAreaViewEdges.TOP)) insets.top else 0.0f
-    val insetRight: Float = if (edges.contains(SafeAreaViewEdges.RIGHT)) insets.right else 0.0f
-    val insetBottom: Float = if (edges.contains(SafeAreaViewEdges.BOTTOM)) insets.bottom else 0.0f
-    val insetLeft: Float = if (edges.contains(SafeAreaViewEdges.LEFT)) insets.left else 0.0f
+    val insetTop: Float = if (edges.top != SafeAreaViewEdgeModes.OFF ) insets.top else 0.0f
+    val insetRight: Float = if (edges.right != SafeAreaViewEdgeModes.OFF) insets.right else 0.0f
+    val insetBottom: Float = if (edges.bottom != SafeAreaViewEdgeModes.OFF) insets.bottom else 0.0f
+    val insetLeft: Float = if (edges.left != SafeAreaViewEdgeModes.OFF) insets.left else 0.0f
     if (localData.mode == SafeAreaViewMode.PADDING) {
-      super.setPadding(Spacing.TOP, insetTop + top)
-      super.setPadding(Spacing.RIGHT, insetRight + right)
-      super.setPadding(Spacing.BOTTOM, insetBottom + bottom)
-      super.setPadding(Spacing.LEFT, insetLeft + left)
+      super.setPadding(Spacing.TOP,  if (edges.top == SafeAreaViewEdgeModes.MAXIMUM ) max(insetTop, top) else insetTop + top)
```

Comment:
It might be easier to make a function like `applyEdges(edgeMode: SafeAreaViewEdgeModes, edgeValue: float, insetValue: float)` - then we can get rid of a fair few lines here